import "library/cpp/actors/protos/actors.proto";
import "ydb/core/protos/base.proto";

package NKikimrTabletPipe;
option java_package = "ru.yandex.kikimr.proto";

enum EFeature {
    FEATURE_GRACEFUL_SHUTDOWN = 1;
}

message TEvConnect {
    optional uint64 TabletId = 1;
    optional NActorsProto.TActorId ClientId = 2;
    optional uint32 Features = 3;
};

message TEvConnectResult {
    optional NKikimrProto.EReplyStatus Status = 1;
    optional uint64 TabletId = 2;
    optional NActorsProto.TActorId ClientId = 3;
    optional NActorsProto.TActorId ServerId = 4;
    optional bool Leader = 5 [default = true];
    optional bool SupportsDataInPayload = 6;
    optional uint32 Generation = 7;
};

message TEvPush {
    optional uint64 TabletId = 1;
    optional uint32 Type = 2;
    optional NActorsProto.TActorId Sender = 3;
    optional bytes Buffer = 4; // may be unset when data is transferred in external payload
    optional uint64 Cookie = 5;
    optional bool ExtendedFormat = 6;
    optional uint64 SeqNo = 7;
};

message TEvPeerClosed {
    optional uint64 TabletId = 1;
    optional NActorsProto.TActorId ClientId = 2;
    optional NActorsProto.TActorId ServerId = 3;
};

message TEvPeerShutdown {
    optional uint64 TabletId = 1;
    optional NActorsProto.TActorId ClientId = 2;
    optional NActorsProto.TActorId ServerId = 3;
    optional uint64 MaxForwardedSeqNo = 4;
};
